state-chain/pallets/cf-funding/src/lib.rs

```diff
@@ -406,6 +413,8 @@ pub mod pallet {
 
 			// If necessary, update account restrictions.
 			if let Some(restricted_balance) = restricted_balances.get_mut(&address) {
+				// We need to keep track of the restricted balance in case the redemption expires.
+				PendingRedemptionsRestrictedBalance::<T>::insert(&account_id, *restricted_balance);
```

Comment:
This is incorrect. It should be `min(debit_amount, restricted_balance)`. 

Imagine the restricted balance is 100 and the user redeems 10 of this and it expires: The new restricted amount would become 190! 

We should add a test case for this.

Reply:
No, we still take the amount in that case. You just check it here in advance and I in the expiry function.

Reply:
Ok... I can see that now. This explains why I was confused by the if/else check below. 

I find this quite confusing: we are storing the entire 'old' restricted balance and then checking during expiry if that old balance was greater than the redemption amount. 

We could just do the check when we process the redemption, and store the amount of restricted funds actually debited. This also makes it easy to see (in polkaJS for example) how much was taken from the account, and how much of that was restricted.

Reply:
agree with dan here. The storage should store the amount of restricted funds in the pending redemption. Easier to  reason and follow what is happening .

Reply:
Agree 👌